meta-mender-tegra/recipes-bsp/u-boot/u-boot-mender-tegra.inc

```diff
@@ -1,19 +1,28 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/patches:"
 
 MENDER_UBOOT_AUTO_CONFIGURE = "0"
-# Use the mmcblk0boot1 partition for uboot environment (partition 2 in uboot)
-MENDER_UBOOT_CONFIG_SYS_MMC_ENV_PART = "2"
+# On tegra devices with a real emmc, Use the mmcblk0boot1 partition for uboot environment (partition 2 in uboot)
+MENDER_UBOOT_CONFIG_SYS_MMC_ENV_PART_tegra186 = "2"
+# On jetson-nano without a real emmc and sd-card instead (no emmc boot partition), use the user area for uboot environment
+MENDER_UBOOT_CONFIG_SYS_MMC_ENV_PART_tegra210 = "0"
 
-# Calculate this offset by adding up the offsets of each partition preceeding the uboot_env partition in sdmmc_boot and aligning to the next
+# For devices with real emmc, calculate this offset by adding up the offsets of each partition preceeding the uboot_env partition in sdmmc_boot and aligning to the next
 # 4096 byte boundary, then subtracting 4 MiB (4194304) since the sdmmc_boot represents the combined boot0 and boot1 partitions
 # Please note the suggestions in the nvidia thread at https://devtalk.nvidia.com/default/topic/1063652/jetson-tx2/mmcblk0boot1-usage-at-address-4177408-and-u-boot-parameter-storage-space-availability/
 # regarding suggestions about locating this environment area in mmcblk0boot1 at 0x3BB000 (3911680)
-MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET = "3911680"
+MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_tegra186 = "3911680"
+
+# For devices with sd-card, u-boot env is in the first partition after the system partitions:
+MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_tegra210 = "12582912"
```

Comment:
@moqmar do you know where this number came from?  I was expecting it to equal the sum of all of the partition offsets in `sdcard-layout-mender.in` from partition 1 through the partition just before the ENV sector but it doesn't look like this is the case?  I think in order for rollback to work I expect it will be necessary for u-boot, mender user space, and the partition layout to all use the same offsets.

Reply:
See https://github.com/Trellis-Logic/meta-mender-community/pull/7#issuecomment-565430070, this answers the question about the offset.

Reply:
@Ecordonnier thanks for the pointer. FYI I can't reference you here in this PR for some reason.

In this case my question is still how this relates to [sdcard-layout-mender.in](https://github.com/Trellis-Logic/meta-mender-community/blob/b91ba8a1a85732d9000b6bf32506664e2b979d22/meta-mender-tegra/recipes-bsp/u-boot/sdcard-layout-mender.in). 
 See [this sheet](https://docs.google.com/spreadsheets/d/1gV8pxmaUPezVQJlbH7V7k_MLYj6tfBoTqWoWNceXjyo/edit?usp=sharing) with offset calculations.  Do you understand how this is resulting in an offset of 12582912/0xC00000 and where the difference comes from?  I think we should ultimately relate these two values in some way through a comment here to make it clear you can't change one without the other, assuming this is the case.

Reply:
Out of curiosity, I compared the values in sdcard-layout-mender.in with the values reported by gdisk. See https://docs.google.com/spreadsheets/d/1yY2XY62qlPzmKpReDVSAY8MVoXLGX2XCnrVG1H813Z0/edit#gid=0

The difference is due to the fact that partitions are aligned to a multiple of 2048 sectors (2048*512 is 1048576 bytes which is bigger than all the partitions before ENV, so the partition offset of ENV 24576 is basically 12*2048), so there is empty space between the partitions which your spreadsheet did not account for. I think the simplest way to find out the offset is to use gdisk, but calculating it manually taking the offset into account is also possible.

Reply:
Regarding the fact that CONFIG_ENV_OFFSET does not need to be hard-coded in a patch, you are right. do_provide_mender_defines exports `CONFIG_ENV_OFFSET=$HEX_MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_1`, and MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_1 has the correct value when compiling u-boot-tegra:
`git/include/config_mender_defines.h:21:#define MENDER_UBOOT_ENV_STORAGE_DEVICE_OFFSET_1 0xc00000`

I will provide a revised patch.

Reply:
Thanks @Ecordonnier
I sent a request to access your spreadsheet link.

Reply:
I had forgotten to make it public, I changed it now.

Reply:
Thanks @Ecordonnier
I get it now, I've updated the sheet at [this link](https://docs.google.com/spreadsheets/d/1gV8pxmaUPezVQJlbH7V7k_MLYj6tfBoTqWoWNceXjyo/edit?usp=sharing) to show how the calculations match between gdisk and sdcard-layout-mender.in